### DISCUSSION TITLE
webserver - serve html

### DISCUSSION THREAD

**Discussion by ppisljar** [tasmota/Berry_playground#36] [Get Help - Q&A]
i uploaded index.html on ufs and want to serve it using berry script, this is my current code:

```
class BerryFS
  def init()
    tasmota.add_driver(self)
  end
  def webrequest()
    import webserver
    try
      var fh = open(webserver.arg('file'),'r')
      # webserver.content_start("brew")
      # webserver.content_send_style()
      webserver.content_response("Content-Type: text/html;\n<html>\n" + fh.read())
      # webserver.content_response(fh.read())
      # webserver.content_stop()
      fh.close()
    except .. as e, m
      # webserver.content_response('error:'..e..' '..m)
      webserver.content_stop()     
    end
  end
  def web_add_handler()
    import webserver
    webserver.on("/fs", / -> self.webrequest(), webserver.HTTP_GET)
  end
end
var berryfs = BerryFS()
```

i am not able to figure out how to send headers correctly, brower will render the html as text. 
i've tried the following:
- just content_response of my index.html
- prepending index.html with headers
- using content_send_style and passing index.html souce as parameter
- using content_Send_Style and content_response with index.html
- using content_start()  (errors out about nil)
- using_content_start("string") chrome errors out with chunking error

thanks everyone for help

**Comment by wjohn007** [comment]
As I see there is by design no way  with webserver to send anything in the header area.

Further I think this will never work 
`   webserver.content_response("Content-Type: text/html;\n<html>\n" + fh.read())`

` webserver.content_response`
This function is according documentation intended for XMLHttpRequest

Take a look to the example in the repository and start with a working example.
https://github.com/arendst/Tasmota/blob/0595130fcf95b9b3cb33322070e01319cc410779/tasmota/berry/examples/web_handler_demo.be

John
